src/main/java/rx/internal/operators/OperatorMulticast.java

```diff
@@ -106,6 +109,21 @@ public void onNext(T args) {
                         subject.onNext(args);
                     }
                 };
+                guardedSubscription = Subscriptions.create(new Action0() {
+                    @Override
+                    public void call() {
+                        Subscription s;
+                        synchronized (guard) {
+                            s = subscription;
```

Comment:
I'd check if ```guardedSubscription == this``` to avoid unsubscribing a newer connection, i.e.:

```java
Subscription s1 = src.connect();
Subscription s2 = src.connect();

s1.unsubscribe();

Subscription s3 = src.connect();

s2.unsubscribe();

// src should be still connected
```